### PR TITLE
Changes to reflect generic -GPUs devices names during training and in…

### PR DESCRIPTION
**Pull request by japarada**
…ference in Monailabel Radiology app loaded in Slicer

### CODE REVIEW

File: monailabel/utils/others/generic.py [hunk 1]
```diff
@@ -307,6 +319,7 @@ def get_bundle_models(app_dir, conf, conf_key="models"):
                 download(name=name, version=version, bundle_dir=model_dir, source=zoo_source, repo=zoo_repo)
                 if version:
                     shutil.move(os.path.join(model_dir, name), p)
+        sys.path.append(p)
```
Comment (SachidanandAlle) [inline monailabel/utils/others/generic.py:322]: why this change?  is your base version little outdated one?

File: monailabel/utils/others/generic.py [hunk 2]
```diff
@@ -101,7 +102,7 @@ def init_log_config(log_config, app_dir, log_file, root_level=None):
         os.makedirs(log_dir, exist_ok=True)
 
         # if not os.path.exists(log_config):
-        shutil.copyfile(default_config, log_config)
+        shutil.copy(default_config, log_config)
```
Comment (SachidanandAlle) [inline monailabel/utils/others/generic.py:105]: why this change?  is your base version little outdated one?

File: monailabel/utils/others/generic.py (hunk 1 above)
Comment (japarada) [reply monailabel/utils/others/generic.py:322]: I see.. I think it did get removed.

File: monailabel/utils/others/generic.py [hunk 3]
```diff
@@ -199,6 +200,17 @@ def device_list():
     return devices
 
 
+def device_list_gpu():
```
Comment (SachidanandAlle) [inline monailabel/utils/others/generic.py:203]: why can't we fix `device_list` method instead of creating new `device_list_gpu`?

can you share the output difference between `device_list` and `device_list_gpu` methods?

File: monailabel/tasks/infer/basic_infer.py [hunk 4]
```diff
@@ -267,12 +267,8 @@ def __call__(
         req = copy.deepcopy(self._config)
         req.update(request)
 
-        # model options
```
Comment (SachidanandAlle) [inline monailabel/tasks/infer/basic_infer.py:270]: some issue with your base version on github.. it has brought in new changes which are not related to this PR

**Comment by SachidanandAlle** [comment]
Thanks for raising the PR.
Can you please provide some context for the issue.  What issue you had to face and how you are able to fix now.

I mean if you can give the old output and new output of device_list method, should be good enough to understand why we needed this change

File: monailabel/tasks/train/basic_train.py [hunk 5]
```diff
@@ -582,7 +582,7 @@ def finalize(self, context):
         early_stop_patience = int(context.request.get("early_stop_patience", 0))
         if early_stop_patience > 0 and context.evaluator:
             kw = self.val_key_metric(context)
-            metric_name = list(kw.keys())[0] if kw else None
+            metric_name = kw.keys()[0] if kw else None
```
Comment (SachidanandAlle) [inline monailabel/tasks/train/basic_train.py:585]: This is some old commit.. not related to this pr

**Comment by SachidanandAlle** [comment]
I believe you wanted some changes in default device for cuda.. what happened?

**Comment by japarada** [comment]
Sorry... I had an issue but I will create PR with changes soon.
Thanks for checking.
